authors/apps/profiles/test_user_profile.py

```diff
@@ -0,0 +1,128 @@
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+    
+    def test_update_user_profile(self):
+        token = self.registration()
+        self.client.post(self.url_register, self.user, format='json')
+        response = self.client.put(self.url_profile,
+        { 
+            'user' : { 
+                'username': 'jDoe', 
+                'bio': 'I am that guy'
+            }
+        }, format='json', HTTP_AUTHORIZATION=token)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_update_profile_short_username_format(self):
+        token = self.registration()
+        self.client.post(self.url_register, self.user, format='json')
+        response = self.client.put(self.url_profile,
+        { 
+            'user' : { 
+                'username': 'jDo', 
+                'bio': 'I am that guy'
+            }
+        }, format='json', HTTP_AUTHORIZATION=token)
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertIn(b'Username must have at least 4 characters', response.content)
+
+    def test_update_profile_no_username(self):
+        token = self.registration()
+        self.client.post(self.url_register, self.user, format='json')
+        response = self.client.put(self.url_profile,
+        { 
+            'user' : { 
+                'username': '', 
+                'bio': 'I am that guy'
+            }
+        }, format='json', HTTP_AUTHORIZATION=token)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertIn(b'This field may not be blank', response.content)
+
+    def test_update_profile_spaced_username(self):
+        token = self.registration()
+        self.client.post(self.url_register, self.user, format='json')
+        response = self.client.put(self.url_profile,
+        { 
+            'user' : { 
+                'username': 'ia o', 
+                'bio': 'I am that guy'
+            }
+        }, format='json', HTTP_AUTHORIZATION=token)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertIn(b'Username cannot have a space', response.content)
+
+    def test_update_profile_blank_bio(self):
+        token = self.registration()
+        self.client.post(self.url_register, self.user, format='json')
+        response = self.client.put(self.url_profile,
+        { 
+            'user' : { 
+                'username': 'iano', 
+                'bio': ''
+            }
+        }, format='json', HTTP_AUTHORIZATION=token)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertIn(b'This field may not be blank', response.content)
+
+    def test_update_profile_spaces_only_in__bio(self):
+        token = self.registration()
+        self.client.post(self.url_register, self.user, format='json')
+        response = self.client.put(self.url_profile,
```

Comment:
the indentation here is a bit off, please make it more readable

Reply:
@FawazFarid I  thought indentation is supposed to be 4 spaces and they are 4 according to how Ian wrote the tests, I would appreciate if you would clarify on this.